### PR TITLE
Create GitHub actions

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
Create GitHub actions for managing GitHub config in an automated fashion.

### CODE REVIEW

File: .github/workflows/tf-merge.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          aws-access-key-id: ${{ secrets.ACTIONS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ACTIONS_SECRET_ACCESS_KEY }}
+          role-to-assume: ${{ secrets.AWS_GHA_ROLE }}
+          role-duration-seconds: 900
+          aws-region: eu-west-2
+
+      - name: "Bootstrap"
+        id: bootstrap
+        run: |
+          python bootstrap.py > ${{ github.workspace }}/${{ github.run_id }}-${{ github.run_number }}.log
```
Comment (connoravo) [inline .github/workflows/tf-merge.yml:44]: Will this become available as an artefact? Do we need to confirm credentials aren't leaked here.

File: bootstrap.py [hunk 2]
```diff
@@ -15,9 +15,7 @@ def main():
     if "AWS_PROFILE" in os.environ:
         secrets_manager = boto3.client("secretsmanager")
     elif "AWS_SECRETS_ROLE" in os.environ:
-        secrets_session = assumed_role_session(os.environ["AWS_SECRETS_ROLE"])
-    else:
-        secrets_manager = secrets_session.client("secretsmanager")
+        secrets_manager = boto3.client("secretsmanager")
```
Comment (steveburtonBUR) [inline bootstrap.py:18]: Can condense this to a single if. Or even better get rid unless there is an else what fails with a useful error (i.e. "Environment variables must contain either AWS_PROFILE or AWS_SECRETS_ROLE"

File: .github/workflows/tf-merge.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          TF_WORKSPACE: default
+          TF_INPUT: false
+          TF_VAR_assume_role: ${var.assume_role}
+          AWS_DEFAULT_REGION: eu-west-2
+
+      - name: "Redact Logs PROD"
+        if: failure()
+        id: redact-logs-mgmt
+        run: |
+              sed -i "s/$ACC_PROD/REDACTED/g" ${{ github.workspace }}/${{ github.run_id }}-${{ github.run_number }}.log
```
Comment (steveburtonBUR) [inline .github/workflows/tf-merge.yml:86]: This looks fine, but this needs to be tested in a real scenario to check the redaction.

File: .github/workflows/tf-merge.yml (hunk 1 above)
Comment (ChrisScottThomas) [reply .github/workflows/tf-merge.yml:44]: Yeah there's a step to redact content.  Creds are hidden as default.

File: bootstrap.py (hunk 2 above)
Comment (steveburtonBUR) [reply bootstrap.py:18]: I'd do an if with both clauses and then the else raise an Error with the above text.

File: bootstrap.py [hunk 4]
```diff
@@ -15,9 +15,9 @@ def main():
     if "AWS_PROFILE" in os.environ:
         secrets_manager = boto3.client("secretsmanager")
     elif "AWS_SECRETS_ROLE" in os.environ:
```
Comment (steveburtonBUR) [inline bootstrap.py:17]:
```suggestion
    if "AWS_PROFILE" in os.environ or "AWS_SECRETS_ROLE" in os.environ:
```

Comment (steveburtonBUR) [inline bootstrap.py:17]:
```suggestion
    if "AWS_PROFILE" in os.environ or "AWS_SECRETS_ROLE" in os.environ:
```

**Review by steveburtonBUR — DISMISSED** [review]

**Review by steveburtonBUR — APPROVED** [review]

**Pull request merged**